### DISCUSSION TITLE
Formatting and style ruleset duplicates

### DISCUSSION THREAD

**Discussion by Lingviston** [detekt/detekt#4390] [Q&A] [answered]
I noticed, that formatting and style rulesets contain duplicating rules. For example: `FinalNewline` in "formatting" and `NewLineAtEndOfFile` in "style"; `MaximumLineLength` in "formatting" and `MaxLineLength` in "style", etc. I guess that the difference is that those from "formatting" ruleset are coming from KtLint, while the "style" ones are native Detekt rules. However, which one am I expected to use if I have choice? Should the documentation highlight duplications like this and explain why it exists?

**Comment by BraisGabin** [comment, marked as answer]
All do the same. The ones from formatting have the option to autofix them.

About which one should you use... That's your call.

About the documentation I agree. I think that the ones from formatting should point out that duplication. The ones from style doesn't need to know the existence of rules that are in a plugin. So, if you want to open an issue adding that to the documentation would be great.

**Reply by cortinico** [reply]
> So, if you want to open an issue adding that to the documentation would be great.

+1 on this. In general we receive several issues/bug-reports of people being confused about this duplication of rules. Having a website page to link would be useful
